napari/plugins/manager.py

```diff
@@ -0,0 +1,337 @@
+import importlib
+import os
+import pkgutil
+import re
+import sys
+from logging import getLogger
+from typing import Dict, Generator, Optional, Tuple, Union
+
+import pluggy
+
+from . import _builtins, hookspecs
+
+logger = getLogger(__name__)
+
+if sys.version_info >= (3, 8):
+    from importlib import metadata as importlib_metadata
+else:
+    import importlib_metadata
+
+
+class PluginError(Exception):
+    def __init__(
+        self, message: str, plugin_name: str, plugin_module: str
+    ) -> None:
+        super().__init__(message)
+        self.plugin_name = plugin_name
+        self.plugin_module = plugin_module
+
+
+class PluginImportError(PluginError):
+    """Raised when a plugin fails to import."""
+
+    def __init__(self, plugin_name: str, plugin_module: str) -> None:
+        msg = f'Failed to import plugin: "{plugin_name}""'
+        super().__init__(msg, plugin_name, plugin_module)
+
+
+class PluginRegistrationError(PluginError):
+    """Raised when a plugin fails to register with pluggy."""
+
+    def __init__(self, plugin_name: str, plugin_module: str) -> None:
+        msg = f'Failed to register plugin: "{plugin_name}""'
+        super().__init__(msg, plugin_name, plugin_module)
+
+
+class NapariPluginManager(pluggy.PluginManager):
+    PLUGIN_ENTRYPOINT = "napari.plugin"
+    PLUGIN_PREFIX = "napari_"
+
+    # for easy availability in try/catch statements without having to import
+    # pluggy ... e.g.: except plugin_manager.PluginValidationError
+    PluginValidationError = pluggy.manager.PluginValidationError
```

Comment:
I'll just remove it altogether.  It was only there so that in _other_ modules, we could access that exception class from our manager instead of having to import from pluggy.  but it's not important